lightning/src/blinded_path/payment.rs

```diff
@@ -35,6 +34,110 @@ use core::ops::Deref;
+}
+
+impl BlindedPaymentPath {
+	/// Create a one-hop blinded path for a payment.
+	pub fn one_hop<ES: Deref, T: secp256k1::Signing + secp256k1::Verification>(
+		payee_node_id: PublicKey, payee_tlvs: ReceiveTlvs, min_final_cltv_expiry_delta: u16,
+		entropy_source: ES, secp_ctx: &Secp256k1<T>
+	) -> Result<(BlindedPayInfo, Self), ()> where ES::Target: EntropySource {
+		// This value is not considered in pathfinding for 1-hop blinded paths, because it's intended to
+		// be in relation to a specific channel.
+		let htlc_maximum_msat = u64::max_value();
+		Self::new(
+			&[], payee_node_id, payee_tlvs, htlc_maximum_msat, min_final_cltv_expiry_delta,
+			entropy_source, secp_ctx
+		)
+	}
+
+	/// Create a blinded path for a payment, to be forwarded along `intermediate_nodes`.
+	///
+	/// Errors if:
+	/// * a provided node id is invalid
+	/// * [`BlindedPayInfo`] calculation results in an integer overflow
+	/// * any unknown features are required in the provided [`ForwardTlvs`]
+	//  TODO: make all payloads the same size with padding + add dummy hops
+	pub fn new<ES: Deref, T: secp256k1::Signing + secp256k1::Verification>(
+		intermediate_nodes: &[ForwardNode], payee_node_id: PublicKey, payee_tlvs: ReceiveTlvs,
+		htlc_maximum_msat: u64, min_final_cltv_expiry_delta: u16, entropy_source: ES,
+		secp_ctx: &Secp256k1<T>
+	) -> Result<(BlindedPayInfo, Self), ()> where ES::Target: EntropySource {
+		let introduction_node = IntroductionNode::NodeId(
+			intermediate_nodes.first().map_or(payee_node_id, |n| n.node_id)
+		);
+		let blinding_secret_bytes = entropy_source.get_secure_random_bytes();
+		let blinding_secret = SecretKey::from_slice(&blinding_secret_bytes[..]).expect("RNG is busted");
+
+		let blinded_payinfo = compute_payinfo(
+			intermediate_nodes, &payee_tlvs, htlc_maximum_msat, min_final_cltv_expiry_delta
+		)?;
+		Ok((blinded_payinfo, Self(BlindedPath {
+			introduction_node,
+			blinding_point: PublicKey::from_secret_key(secp_ctx, &blinding_secret),
+			blinded_hops: blinded_hops(
+				secp_ctx, intermediate_nodes, payee_node_id, payee_tlvs, &blinding_secret
+			).map_err(|_| ())?,
+		})))
+	}
+
+	/// Returns the introduction [`NodeId`] of the blinded path, if it is publicly reachable (i.e.,
+	/// it is found in the network graph).
+	pub fn public_introduction_node_id<'a>(
+		&self, network_graph: &'a ReadOnlyNetworkGraph
+	) -> Option<&'a NodeId> {
+		self.0.public_introduction_node_id(network_graph)
+	}
+
+	/// The [`IntroductionNode`] of the blinded path.
+	pub fn introduction_node(&self) -> &IntroductionNode {
+		&self.0.introduction_node
+	}
+
+	/// Used by the [`IntroductionNode`] to decrypt its [`encrypted_payload`] to forward the payment.
+	///
+	/// [`encrypted_payload`]: BlindedHop::encrypted_payload
+	pub fn blinding_point(&self) -> PublicKey {
+		self.0.blinding_point
+	}
+
+	/// The [`BlindedHop`]s within the blinded path.
+	pub fn blinded_hops(&self) -> &[BlindedHop] {
+		&self.0.blinded_hops
+	}
+
+	#[cfg(any(test, fuzzing))]
```

Comment:
Keeping it private would at least prevent someone from forming a `BlindedPaymentPath` from the data inside of a `BlindedMessagePath`, which would be have the wrong encrypted data.

Reply:
If we care to go this way, the only reason I can think of is for creating a blinded payment path to be received over by another implementation. If that's the case, it would make more sense for the API to only take a raw payload for the last hop, not intermediate hops, and to still construct its own payinfo IMO. Maybe not too high priority, though?